### PR TITLE
Simplify control system observations

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Uses https://github.com/sxs-collaboration/spectre/pull/3873 to simplify observing for the control system.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ControlSystem/WriteData.hpp [hunk 1]
```diff
@@ -157,31 +109,26 @@ void write_components_to_disk(
   // ControlSystems/Translation/Y.dat
   // ControlSystems/Translation/Z.dat
   const size_t num_components = function_at_current_time[0].size();
+  const auto& legend = WriterHelper::legend;
```
Comment (nilsvu) [inline src/ControlSystem/WriteData.hpp]: Just inline the legend here and delete the `WriterHelper`

**Review by nilsvu — CHANGES_REQUESTED** [review]
squash directly

File: src/ControlSystem/WriteData.hpp (hunk 1 above)
Comment (nilsvu) [reply src/ControlSystem/WriteData.hpp]: Oh I meant: just write it right here in line 112, and `std::move` it into the simple action invocation below. It's taken by value by the action anyway.

Comment (knelli2) [reply src/ControlSystem/WriteData.hpp]: Ah ok my bad I misunderstood. Thanks for clarifying :)

**Review by nilsvu — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**